pkg/util/cluster/cluster.go

```diff
@@ -447,6 +447,7 @@ func (c *Cluster) createCluster(ctx context.Context, vnetResourceGroup, clusterN
 				ResourceGroupID:      fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", c.env.SubscriptionID(), "aro-"+clusterName),
 				FipsValidatedModules: api.FipsValidatedModulesEnabled,
 				Version:              osClusterVersion,
+				PullSecret:           api.SecureString(os.Getenv("USER_PULL_SECRET")),
```

Comment:
Can we include an accompanying docs change in this PR?

We can update this one with instructions on passing in a pull secret: https://github.com/Azure/ARO-RP/blob/master/docs/deploy-development-rp.md#run-the-rp-and-create-a-cluster

Reply:
I don't think we need it, because `USER_PULL_SECRET` is already configured in the secret, so we don't need to set it explicitly.

And also it is not meant as a general way of configuring the pull-secret.
If we want to use other pull-secret then we can set it after creation, so we don't need to use this `USER_PULL_SECRET` variable. 
It is only needed when we want to use a installer image stored in other place than ACR.